tests/storage/storage.go

```diff
@@ -1452,14 +1452,27 @@ func deletePvAndPvc(name string) {
 	}
 }
 
+func runPodAndExpectCompletion(pod *k8sv1.Pod) *k8sv1.Pod {
+	virtClient := kubevirt.Client()
+
+	var err error
+	pod, err = virtClient.CoreV1().Pods(testsuite.GetTestNamespace(pod)).Create(context.Background(), pod, metav1.CreateOptions{})
+	Expect(err).ToNot(HaveOccurred())
+	Eventually(ThisPod(pod), 120).Should(BeInPhase(k8sv1.PodSucceeded))
+
+	pod, err = ThisPod(pod)()
+	Expect(err).ToNot(HaveOccurred())
```

Comment:
Two comments about this:
- I suspect this is not safe. Although probably rare, if the pod no longer exists, it will return no error and the pod will be nil.
- A possible solution, which seems safe to me, is to change `ThisPod` such that it will update `pod` if successfully fetched. Then, we will not need this addition at all.

Reply:
`ThisPod` function is consistent with all other ThisX functions non of which update the object. I added a check here that the returned pod is not nil.